pulsar-io/kinesis/src/main/java/org/apache/pulsar/io/kinesis/KinesisSink.java

```diff
@@ -121,10 +130,8 @@ public void write(Record<byte[]> record) throws Exception {
                 ? partitionedKey.substring(0, maxPartitionedKeyLength - 1)
                 : partitionedKey; // partitionedKey Length must be at least one, and at most 256
         ByteBuffer data = createKinesisMessage(kinesisSinkConfig.getMessageFormat(), record);
-        ListenableFuture<UserRecordResult> addRecordResult = kinesisProducer.addUserRecord(this.streamName,
-                partitionedKey, data);
-        addCallback(addRecordResult,
-                ProducerSendCallback.create(this, record, System.nanoTime()), directExecutor());
+        Backoff backoff = new Backoff(1, TimeUnit.SECONDS, 1, TimeUnit.MINUTES, 0, TimeUnit.SECONDS);
```

Comment:
Do we need to create a Backoff instance for each data write process? We can just reuse the object by using the reset() method to reset the backoff?

Reply:
And it's better to make the initial timeout and max timeout configuable

Reply:
Thanks! I have fixed it.